src/main/java/com/imgix/URLHelper.java

```diff
@@ -158,6 +161,58 @@ private static String joinList(List<String> strings, String separator) {
     return sb.toString();
   }
 
+  // checkProxyStatus checks if the path has one of the four possible
+  // acceptable proxy prefixes. First we check if the path has the
+  // correct ascii prefix. If it does then we know that it is a proxy,
+  // but it's not percent encoded. Second, we check if the path is
+  // prefixed by a percent-encoded prefix. If it is, we know that it's
+  // a proxy and that it's percent-encoded. Finally, if the path isn't
+  // prefixed by any of these four prefixes, it is not a valid proxy.
+  // This might be "just enough validation," but if we run into issues
+  // we can make this check smarter/more-robust.
+  public static Map<String, Boolean> checkProxyStatus(String p) {
+    Map<String, Boolean> status = new HashMap<String, Boolean>();
+    String path = p;
+    path.replaceAll("^/", "");
+
+    String asciiHTTP = "http://";
+    String asciiHTTPS = "https://";
+
+    String encodedHTTP = "http%3A%2F%2F";
+    String encodedHTTPS = "https%3A%2F%2F";
+
+    String encodedHTTPLower = "http%3a%2f%2f";
+    String encodedHTTPSLower = "https%3a%ff%2f";
```

Comment:
Same comment applies here as https://github.com/imgix/imgix-java/pull/68#discussion_r606312408

Reply:
Ms, this was pulled directly from [imgix-go](https://github.com/imgix/imgix-go/blob/30f9e8976bddcc02fc21f0accbb341ec0d69f226/v2/encoding.go#L12-L46) for what it’s worth. Can remove as well so we can discuss more.

Reply:
[Removed](https://github.com/imgix/imgix-java/pull/70/commits/12b5efd0c90f50bdbe281748198fbe3b5424517e) them until we can discuss

Reply:
Turns out this was always supported in the library, so we're keeping it for now.